### PR TITLE
chore(storybook): upgrade storybook

### LINKED ISSUE #29530: Upgrade StoryBook
### Parent Issue

https://github.com/dotCMS/core/issues/29058

### Objective

The stories of our components use a StoryBook older version and there are some stories broken and this is important as part of the PrimeNG upgrade to V7 to check that the components work correctly with this migration as a result this task is to fix the broken storybooks and upgrade using the latest StoryBook API.

### PR DESCRIPTION
**Pull request by nicobytes**
### Parent Issue

https://github.com/dotCMS/core/issues/29530

### Objective

The stories of our components use a StoryBook older version and there are some stories broken and this is important as part of the PrimeNG upgrade to V17 to check that the components work correctly with this migration as a result this task is to fix the broken storybooks and upgrade using the latest StoryBook version.

### Proposed Changes
* Upgrade storybook from v7 to v8
* `Story` was removed in v8 and replace it for `StoryObj`
* Use new format to write stories with strong typing
* Support to signals inputs in stories
* 8 Fixed stories (WIP)

### Checklist
- [x] Tests
- [x] Translations
- [x] Security Implications Contemplated (add notes if applicable)

### Fixed Stories
Original             |  Updated
:-------------------------:|:-------------------------:
localhost_4400__path=_docs_dotcms-content-palette--docs  |  localhost_4401__path=_docs_dotcms-content-palette--docs
localhost_4400__path=_docs_dotcms-content-palette--docs | localhost_4401__path=_docs_dotcms-content-palette--docs
localhost_4400__path=_docs_primeng-form-treeselect--docs |  localhost_4401__path=_docs_dotcms-content-palette--docs (1)
localhost_4400__path=_docs_primeng-form-treeselect--docs (1) | localhost_4401__path=_docs_dotcms-content-palette--docs (2)
localhost_4401__path=_docs_dotcms-device-selector-seo--docs | localhost_4400_
localhost_4401__path=_docs_dotcms-device-selector-seo--docs (2) | localhost_4400__path=_docs_dotcms-results-seo-tool--docs

This PR fixes: #29530

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/stories/Introduction.stories.mdx [hunk 1]
```diff
@@ -1,18 +0,0 @@
-import { Meta } from '@storybook/addon-docs/blocks';
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/stories/Introduction.stories.mdx:1]: why removed ?

File: core-web/apps/dotcms-ui/src/stories/primeng/media/Galleria.stories.ts [hunk 2]
```diff
@@ -1,23 +1,11 @@
-import { Meta, StoryFn } from '@storybook/angular';
+import { faker } from '@faker-js/faker';
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/stories/primeng/media/Galleria.stories.ts:1]: nice

**Review by hmoreras — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-categories/dot-categories-list/dot-categories-list.component.ts [hunk 3]
```diff
@@ -16,7 +16,9 @@ import { DotCategoriesListState, DotCategoriesListStore } from './store/dot-cate
     providers: [DotCategoriesListStore]
 })
 export class DotCategoriesListComponent {
-    vm$: Observable<DotCategoriesListState> = this.store.vm$;
+    readonly #store = inject(DotCategoriesListStore);
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-categories/dot-categories-list/dot-categories-list.component.ts:19]: Why modify this if is an upgrade of storybook?

File: core-web/apps/dotcms-ui/src/stories/Introduction.stories.mdx (hunk 1 above)
Comment (nicobytes) [reply core-web/apps/dotcms-ui/src/stories/Introduction.stories.mdx:1]: The mdx does not work properly

File: core-web/apps/dotcms-ui/src/app/portlets/dot-categories/dot-categories-list/dot-categories-list.component.ts (hunk 3 above)
Comment (nicobytes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-categories/dot-categories-list/dot-categories-list.component.ts:19]: Was mistaken

**Pull request merged**